dnadoh/synthesize.py

```diff
@@ -296,25 +327,40 @@ def main():
     options = parse_args()
     random.seed(options.seed)
     genomes, people = generate(options)
-    _write(options.output_stem, genomes, people)
+    if options.output_stem:
+        _write_files(options, genomes, people)
+    else:
+        _write_summary(sys.stdout, options, genomes, people)
 
 
 def parse_args():
     """Get command-line arguments."""
     parser = argparse.ArgumentParser()
-    parser.add_argument("--length", type=int, help="genome length")
-    parser.add_argument("--num_genomes", type=int, help="number of genomes (people)")
+    parser.add_argument("--parameters", type=str, default=None, help="JSON parameter file")
+    parser.add_argument("--length", type=int, default=None, help="genome length")
+    parser.add_argument("--num_genomes", type=int, default=None, help="number of genomes (people)")
     parser.add_argument(
-        "--num_mutations", type=int, help="number of significant mutations"
+        "--num_mutations", type=int, default=None, help="number of significant mutations"
     )
     parser.add_argument(
         "--max_num_other_mutations",
         type=int,
+        default=None,
         help="maximum number of other mutations per person",
     )
-    parser.add_argument("--seed", type=int, help="RNG seed")
-    parser.add_argument("--output_stem", type=str, help="output path/file stem")
-    return parser.parse_args()
+    parser.add_argument("--seed", type=int, default=None, help="RNG seed")
+    parser.add_argument("--output_stem", type=str, default=None, help="output path/file stem")
+
+    options = parser.parse_args()
+    if options.parameters:
+        assert (options.length is None) and (options.num_genomes is None) and (options.num_mutations is None) and (options.max_num_other_mutations is None), "Cannot specify individual parameters and parameter file"
+        with open(options.parameters, "r") as reader:
+            parameters = json.load(reader)
+            for key in parameters:
+                setattr(options, key, parameters[key])
+    else:
+        assert options.seed is not None, "Must specify random number seed"
```

Comment:
you've checked that seed isn't None when the parameters json isn't used. But what if a json is passed in, but "seed" isn't in that json?